Completed HW 23.1 and 24.1

RomanMinevich opened the pull request.

Commit:
Completed HW 23.1 and 24.1

Commit:
Added PreparedStatement.RETURN_GENERATED_KEYS operator to jdbc package classes

boroda4436 commented on src/main/java/mate.academy.internetshop/controller/BucketController.java:
```diff
@@ -25,7 +25,7 @@
             throws ServletException, IOException {
         Long userId = (Long)request.getSession(true).getAttribute("userId");
-        request.setAttribute("items", bucketService.getByUserId(userId).getItems());
+        request.setAttribute("items", bucketService.get(userId).getItems());
```
@RomanMinevich Це не корректно передавати userId в метод get. Очікується що сюди потрібно передавати bucketId. Попередня версія з getByUserId більш корректно буде виглялдати

RomanMinevich replied:
Виправив

boroda4436 commented on src/main/java/mate.academy.internetshop/dao/RoleDao.java:
```diff
@@ -0,0 +1,12 @@
+
+    Role get(Long id);
+
+    Set<Role> getUserRoles(Long userId);
```
@RomanMinevich Думаю що цей метод було б краще використати в userDao, а ще краще витягувати ролі прямо з user'a

RomanMinevich replied:
Переніс даний метод до UserDao і додав змінну Set<Role> roles до User (отримати перелік ролей користувача тепер можна "прямо з user'a")

boroda4436 commented on src/main/java/mate.academy.internetshop/dao/UserDao.java:
```diff
@@ -17,7 +16,7 @@
     User login(String phoneNumber, String password) throws AuthenticationException;
 
-    Optional<User> getByToken(String token);
+    User getByToken(String token);
```
@RomanMinevich А чому прибрав Optional?

RomanMinevich replied:
За непотрібнісію) 
![Снимок экрана 2019-10-07 в 15 05 07](https://user-images.githubusercontent.com/52323566/66310197-3fc4f180-e914-11e9-8761-9fdf2f7f3d0d.png)
–––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––
![Снимок экрана 2019-10-07 в 15 06 17](https://user-images.githubusercontent.com/52323566/66310198-3fc4f180-e914-11e9-995f-b01b9e6e3acc.png)

Commit:
Changed solution for HW 23.1 – 24.1:
Added variable Set<Role> roles to User model and private method setRoles(User user) to UserDao
Created RoleService interface and its implementation. Removed Enum RoleName from Role model, removed method getRolesForUser(Long userId) from RoleDao, added method get(String name) to RoleDao and RoleService
Changed Map<> urls in AuthorizationFilter parameters to <String, Role>
Added method get(Long userId) to BucketService

Commit:
Changed solution for HW 23.1 – 24.1:
Added variable Set<Role> roles to User model and private method setRoles(User user) to UserDao
Created RoleService interface and its implementation. Removed Enum RoleName from Role model, removed method getRolesForUser(Long userId) from RoleDao, added method get(String name) to RoleDao and RoleService
Changed Map<> urls in AuthorizationFilter parameters to <String, Role>
Added method get(Long userId) to BucketService
